pkg/fqdn/namemanager/manager_test.go

```diff
@@ -195,6 +226,212 @@ func TestNameManagerIPCacheUpdates(t *testing.T) {
+
+	// Run GC and check that the prefix is upserted to the ipcache
+	require.NoError(t, nameManager.doGC(t.Context()))
+
+	// Ensure all IPcache operations done by GC have been processed
+	err = ipc.WaitForRevision(t.Context(), ipc.UpsertMetadataBatch())
+	require.NoError(t, err)
+
+	id, found = ipc.LookupByPrefix(prefixTwo.String())
+	require.True(t, found)
+	ident = ipc.IdentityAllocator.LookupIdentityByID(t.Context(), id.ID)
+	require.Equal(t, labels.FromSlice([]labels.Label{ciliumIOSel.IdentityLabel(), labels.ParseLabel("reserved:world-ipv4")}), ident.Labels)
+	require.Contains(t, nameManager.cache.LookupIP(prefixTwo.AsPrefix().Addr()), dns.FQDN("cilium.io"))
+
+	// Add another prefix to local cache
+	ep.DNSHistory.Update(lookupTime, dns.FQDN("cilium.io"), []netip.Addr{prefixThree.AsPrefix().Addr()}, 10)
+
+	// Run GC to ensure the IP<>name mapping is not added to the global cache yet
+	require.NoError(t, nameManager.doGC(t.Context()))
+
+	// Ensure all IPcache operations done by GC have been processed
+	err = ipc.WaitForRevision(t.Context(), ipc.UpsertMetadataBatch())
+	require.NoError(t, err)
+
+	require.Empty(t, nameManager.cache.LookupIP(prefixThree.AsPrefix().Addr()))
+
+	_, found = ipc.LookupByPrefix(prefixThree.String())
+	require.False(t, found)
+	require.NotContains(t, nameManager.cache.LookupIP(prefixThree.AsPrefix().Addr()), dns.FQDN("cilium.io"))
+
+	// Upsert to global cache and ensure its present
+	<-nameManager.UpdateGenerateDNS(t.Context(), lookupTime, dns.FQDN("cilium.io"), &fqdn.DNSIPRecords{TTL: 10, IPs: []netip.Addr{prefixThree.AsPrefix().Addr()}}, ep.DNSHistory)
+
+	id, found = ipc.LookupByPrefix(prefixThree.String())
+	require.True(t, found)
+	ident = ipc.IdentityAllocator.LookupIdentityByID(t.Context(), id.ID)
+	require.Equal(t, labels.FromSlice([]labels.Label{ciliumIOSel.IdentityLabel(), labels.ParseLabel("reserved:world-ipv4")}), ident.Labels)
+	require.Contains(t, nameManager.cache.LookupIP(prefixThree.AsPrefix().Addr()), dns.FQDN("cilium.io"))
+
+	// Run GC and ensure it's still present
+	require.NoError(t, nameManager.doGC(t.Context()))
+
+	// Ensure all IPcache operations done by GC have been processed
+	err = ipc.WaitForRevision(t.Context(), ipc.UpsertMetadataBatch())
+	require.NoError(t, err)
+
+	id, found = ipc.LookupByPrefix(prefixThree.String())
+	require.True(t, found)
+	ident = ipc.IdentityAllocator.LookupIdentityByID(t.Context(), id.ID)
+	require.Equal(t, labels.FromSlice([]labels.Label{ciliumIOSel.IdentityLabel(), labels.ParseLabel("reserved:world-ipv4")}), ident.Labels)
+
+	// Lock endpoint manager to freeze GC in time
+	epMgr.mu.Lock()
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		require.NoError(t, nameManager.doGC(t.Context()))
+
+		// Ensure all IPcache operations done by GC have been processed
+		err = ipc.WaitForRevision(t.Context(), ipc.UpsertMetadataBatch())
+		require.NoError(t, err)
+	}()
```

Comment:
Note: changed this from `sync.WaitGroup.Go` to this construct (essentially inlining that method) because it is not available in Go 1.24 that we are using on `v1.18`.